tests/tests/storage_read_replay.rs

```diff
@@ -0,0 +1,178 @@
+use fuel_core::service::{
+    Config,
+    FuelService,
+};
+use fuel_core_client::client::{
+    types::TransactionStatus,
+    FuelClient,
+};
+use fuel_core_types::{
+    fuel_asm::{
+        op,
+        GTFArgs,
+        RegId,
+    },
+    fuel_tx::{
+        Bytes32,
+        ContractId,
+        CreateMetadata,
+        Finalizable,
+        Input,
+        Output,
+        StorageSlot,
+        TransactionBuilder,
+    },
+    fuel_types::BlockHeight,
+    fuel_vm::{
+        Salt,
+        SecretKey,
+    },
+};
+use futures::StreamExt;
+use rand::{
+    Rng,
+    SeedableRng,
+};
+
+async fn make_counter_contract(
+    client: &FuelClient,
+    rng: &mut rand::rngs::StdRng,
+) -> (ContractId, BlockHeight) {
+    let maturity = Default::default();
+
+    let code: Vec<_> = [
+        // Make zero key
+        op::movi(0x12, 32),
+        op::aloc(0x12),
+        // Read value
+        op::srw(0x10, 0x11, 0x12),
+        // Increment value
+        op::addi(0x10, 0x10, 1),
+        // Write value
+        op::sww(0x12, 0x11, 0x10),
+        // Return new counter value
+        op::ret(0x10),
+    ]
+    .into_iter()
+    .collect();
+
+    let salt: Salt = rng.gen();
+    let tx = TransactionBuilder::create(
+        code.into(),
+        salt,
+        vec![StorageSlot::new(Bytes32::zeroed(), Bytes32::zeroed())],
+    )
+    .maturity(maturity)
+    .add_fee_input()
+    .add_contract_created()
+    .finalize();
+
+    let contract_id = CreateMetadata::compute(&tx).unwrap().contract_id;
+
+    let mut status_stream = client.submit_and_await_status(&tx.into()).await.unwrap();
```

Comment:
can we use `submit_and_await_commit_with_tx` here instead?
https://github.com/FuelLabs/fuel-core/blob/bb0368e262ce3793cff3990dcc92c11354f2dc89/crates/client/src/client.rs#L569-L570

it'll need the `subscriptions` feature to be enabled, but i think it might be already

Reply:
Good catch, done in fc097bf56e